### PR TITLE
IA 188 rewriting required (Store)

### PR DESCRIPTION
**Pull request by AlexWeinstein92**
Rewriting required fields to use default Options again in Store service

### CODE REVIEW

**Review by dag-yoppworks — CHANGES_REQUESTED** [review]
Am I missing where the validation has moved to now? It looks like now we will just get exceptions in the service if the message has missing data. I would really prefer to see that modeled explicitly somehow.

File: store/src/main/scala/com/improving/app/store/domain/Store.scala [hunk 1]
```diff
@@ -291,55 +308,71 @@ object Store {
...
-        name = fieldsToUpdate.name
-          .getOrElse(stateInfo.name),
-        description = fieldsToUpdate.description
-          .getOrElse(stateInfo.description),
-        sponsoringOrg = Some(
-          fieldsToUpdate.sponsoringOrg
-            .getOrElse(stateInfo.getSponsoringOrg)
-        )
-      )
+      val updatedInfo = command.newInfo match {
```
Comment (dag-yoppworks) [inline store/src/main/scala/com/improving/app/store/domain/Store.scala:331]: this match is idiomatically written `.map`, that is:
```
maybeA match {
  case Some(a) => Some(f(a))
  case None => None
}
```
becomes `maybeA.map(f)`

File: store/src/main/scala/com/improving/app/store/domain/Store.scala [hunk 2]
```diff
@@ -291,55 +308,71 @@ object Store {
...
+          )
+        case None => None
+      }
       val updatedMetaInfo = updateMetaInfo(Some(state.metaInfo), command.onBehalfOf)
 
       Right(
         StoreInfoEdited(
           storeId = command.storeId,
-          info = Some(StoreOrEditableInfo(StoreOrEditableInfo.InfoOrEditable.Info(updatedInfo))),
+          info = Some(StoreOrEditableInfo(StoreOrEditableInfo.InfoOrEditable.Info(updatedInfo match {
```
Comment (dag-yoppworks) [inline store/src/main/scala/com/improving/app/store/domain/Store.scala:354]: this could be idiomatically written `updatedInfo.getOrElse(state.info)`

**Review by dag-yoppworks — CHANGES_REQUESTED** [review]
Looks good, I just see a couple places where we could be using Option more idiomatically

File: store/src/main/scala/com/improving/app/store/domain/Store.scala [hunk 3]
```diff
@@ -328,33 +328,28 @@ object Store {
...
-            StoreInfo(
-              name = fieldsToUpdate.name
-                .getOrElse(stateInfo.name),
-              description = fieldsToUpdate.description
-                .getOrElse(stateInfo.description),
-              sponsoringOrg = Some(
-                fieldsToUpdate.sponsoringOrg
-                  .getOrElse(stateInfo.getSponsoringOrg)
-              )
+      val updatedInfo = command.newInfo.flatMap { fieldsToUpdate =>
```
Comment (dag-yoppworks) [inline store/src/main/scala/com/improving/app/store/domain/Store.scala:331]: Option `flatMap` is typically used to chain optional values together, if you need something like `Option[Option[A]] => Option[B]`, while `map` is used to transform the contents of an Option like you have here, `Option[A] => Option[B]`.

https://freecontent.manning.com/using-option-in-scala-part-2-map-and-flatmap/

Sometimes I find it helpful to think of Option as a container with 0 or 1 elements for the purposes of these methods.

**Review by dag-yoppworks — CHANGES_REQUESTED** [review]
One more nitpick about the way you've used Option

File: store/src/main/scala/com/improving/app/store/domain/Store.scala (hunk 3 above)
Comment (AlexWeinstein92) [reply store/src/main/scala/com/improving/app/store/domain/Store.scala:331]: The return object was wrapped in Some som I used flatMap. Didn't look closely enough. Fixed now

**Review by dag-yoppworks — APPROVED** [review]
Alright, thanks! Looks good.

**Pull request merged**